src/components/Pet.jsx

```diff
@@ -0,0 +1,154 @@
+import React, { useEffect, useState } from "react";
```

Comment:
We should organize this differently to avoid having ifs and ternarries to decide if it is edit or create.
We could have this a a pet form component, that doesn't know if it is edit or create form and receives a `initialValues` and `handleSubmit` props. Then we have a EditForm and a CreateForm components which render the generic pet form and pass the appropriate initial values / submit function/